new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -494,6 +738,189 @@ void tearDown() {
     assertDoesNotThrow(() -> deleteDomainCustomResource(domainUid, domainNamespace1),
             "deleteDomainCustomResource failed with ApiException");
     logger.info("Deleted Domain Custom Resource " + domainUid + " from " + domainNamespace1);
+
+    // delete the domain images created in the test class
+    if (miiImagePatchAppV2 != null) {
+      deleteImage(miiImagePatchAppV2);
+    }
+    if (miiImageAddSecondApp != null) {
+      deleteImage(miiImageAddSecondApp);
+    }
+  }
+
+  private void pushImageIfNeeded(String image) {
+    // push the image to a registry to make the test work in multi node cluster
+    if (!REPO_USERNAME.equals(REPO_DUMMY_VALUE)) {
+      logger.info("docker login to registry {0}", REPO_REGISTRY);
+      assertTrue(dockerLogin(REPO_REGISTRY, REPO_USERNAME, REPO_PASSWORD), "docker login failed");
+
+      logger.info("docker push image {0} to registry {1}", image, REPO_REGISTRY);
+      assertTrue(dockerPush(image), String.format("docker push failed for image %s", image));
+    }
+  }
+
+  private String createUniqueImageTag() {
+    // create unique image name with date
+    DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    Date date = new Date();
+    return dateFormat.format(date) + "-" + System.currentTimeMillis();
+  }
+
+  private String createImageName(String baseImageName) {
+    // Add repository name in image name for Jenkins runs
+    return REPO_USERNAME.equals(REPO_DUMMY_VALUE) ? baseImageName : REPO_NAME + baseImageName;
```

Comment:
This is what is breaking your test. You need to add REPO_NAME all of the time.

Reply:
REPO_NAME will be empty when no push is required.  It is no longer tied to whether or not REPO_USERNAME is set.

Reply:
fixed.